From 731ad599015e0ee9534d90cd2679d596a917c2cf Mon Sep 17 00:00:00 2001
From: Venkatasaiprudhvi Raj Sannidhi <vsannidhi@maxlinear.com>
Date: Wed, 3 Sep 2025 04:50:58 -0700
Subject: [PATCH] WLANRTSYS-91000 [OCT][Autococ]The OMN IE channel width is set
 to 80 MHz, while the 5G and 6Gchannel width is 160 MHz during traffic

Issue: When AutoCOC, BW switch is enabled, once the NSS moved from 1x1 to 4x4, BW is switched to 160MHz but OMNIE in beacon advertises as 80MHz BW.

Root case: When the AutoCOC switches BW back to 160 MHz, the OMN IE will
be reconstructed with the new BW as part of the function "mxl_hostapd_eid_opmodenotif".
Here the BW bit setting is done incorrectly for 160MHz.

Fix: As per spec, in the channel width field of OMN IE, BIT 0 and BIT 1 should be set for 160MHz case.
---
 src/ap/mxl_hostapd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ap/mxl_hostapd.c b/src/ap/mxl_hostapd.c
index 209fd976c..ed7c6eca6 100644
--- a/src/ap/mxl_hostapd.c
+++ b/src/ap/mxl_hostapd.c
@@ -2602,7 +2602,7 @@ u8 * mxl_hostapd_eid_opmodenotif(struct hostapd_data *hapd, u8 *eid)
 			*pos |= BIT(1);
 			break;
 		case CONF_OPER_CHWIDTH_160MHZ:
-			*pos |=  BIT (1) | BIT(2);
+			*pos |=  BIT (0) | BIT(1);
 			break;
 		default:
 			*pos &= ~BIT(0);
-- 
2.43.0

